File: ToDoModel/Dimension Libraries/Project/ToDoProjectInfoUtilities.vb
Imports System.IO
Imports System.Runtime.Serialization.Formatters.Binary

Public Module ToDoProjectInfoUtilities
    Public Function LoadToDoProjectInfo(filePath As String) As ProjectInfo
        Dim ProjInfo As ProjectInfo
        If File.Exists(filePath) Then
            'Dim TestFileStream As Stream = File.OpenRead(filePath)
            Dim deserializer As New BinaryFormatter
            Try
                Using TestFileStream As Stream = File.OpenRead(filePath)
                    ProjInfo = CType(deserializer.Deserialize(TestFileStream), ProjectInfo)
                End Using
            Catch ex As UnauthorizedAccessException
                MsgBox("Unexpected Access Error. Duplicate Instance Running?")
                Throw ex
            Catch ex As IOException
                MsgBox("Unexpected IO Error. Is Project Info File Corrupt?")
                Throw ex
            End Try

            ProjInfo.FileName = filePath
            ProjInfo.Sort()
            Return ProjInfo
        Else
            ProjInfo = New ProjectInfo
            ProjInfo.Save(filePath)
            Return ProjInfo
        End If

    End Function
End Module

File: ToDoModel/Dimension Libraries/Project/ToDoProjectInfoUtilities.cs
using System;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;
using System.Windows.Forms;
using ToDoModel.Legacy;

namespace ToDoModel
{

    public static class ToDoProjectInfoUtilities
    {
        public static ProjectInfoLegacy LoadToDoProjectInfo(string filePath)
        {
            ProjectInfoLegacy ProjInfo;
            if (File.Exists(filePath))
            {
                // Dim TestFileStream As Stream = File.OpenRead(filePath)
                var deserializer = new BinaryFormatter();
                try
                {
                    using (Stream TestFileStream = File.OpenRead(filePath))
                    {
                        ProjInfo = (ProjectInfoLegacy)deserializer.Deserialize(TestFileStream);
                    }
                }
                catch (UnauthorizedAccessException ex)
                {
                    MessageBox.Show("Unexpected Access Error. Duplicate Instance Running?");
                    throw ex;
                }
                catch (IOException ex)
                {
                    MessageBox.Show("Unexpected IO Error. Is Project Info File Corrupt?");
                    throw ex;
                }

                ProjInfo.FileName = filePath;
                ProjInfo.Sort();
                return ProjInfo;
            }
            else
            {
                ProjInfo = new ProjectInfoLegacy();
                ProjInfo.Save(filePath);
                return ProjInfo;
            }

        }
    }
}